mathext/ell_complete_test.go

```diff
@@ -0,0 +1,27 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+	"testing"
+)
+
+// TestCompleteKE checks if the Legendre's relation for m=0.0001(0.0001)0.9999
+// is satisfied with accuracy 1e-14.
+func TestCompleteKE(t *testing.T) {
+	const tol = 1.0e-14
+
+	for m := 1; m <= 9999; m++ {
+		mf := float64(m) / 10000
+		mp := 1 - mf
+		K, Kp := CompleteK(mf), CompleteK(mp)
+		E, Ep := CompleteE(mf), CompleteE(mp)
+		legendre := math.Abs(E*Kp + Ep*K - K*Kp - math.Pi/2)
+		if legendre > tol {
+			t.Errorf("legendre > tol: m=%v, legendre=%v, tol=%v", mf, legendre, tol)
```

Comment:
In the case of failure, this can spew up to 1e4 log lines. Make this a `t.Fatalf` and note in the output the `m` at which the failure happened.